server/service/microsoft_mdm.go

```diff
@@ -572,6 +784,239 @@ func (svc *Service) GetMDMWindowsPolicyResponse(ctx context.Context, authToken s
+}
+
+// storeWindowsMDMEnrolledDevice stores the device information to the list of MDM enrolled devices
+func (svc *Service) storeWindowsMDMEnrolledDevice(ctx context.Context, secTokenMsg *fleet.RequestSecurityToken) error {
+	const (
+		error_tag = "windows MDM enrolled storage: "
+	)
+
+	// Getting the DeviceID context information from the RequestSecurityToken msg
+	reqDeviceID, err := GetContextItem(secTokenMsg, mdm.ReqSecTokenContextItemDeviceID)
+	if err != nil {
+		return fmt.Errorf("%s %v", error_tag, err)
+	}
+
+	// Getting the HWDevID context information from the RequestSecurityToken msg
+	reqHWDevID, err := GetContextItem(secTokenMsg, mdm.ReqSecTokenContextItemHWDevID)
+	if err != nil {
+		return fmt.Errorf("%s %v", error_tag, err)
+	}
+
+	// Getting the Enroll DeviceType context information from the RequestSecurityToken msg
+	reqDeviceType, err := GetContextItem(secTokenMsg, mdm.ReqSecTokenContextItemDeviceType)
+	if err != nil {
+		return fmt.Errorf("%s %v", error_tag, err)
+	}
+
+	// Getting the Enroll DeviceName context information from the RequestSecurityToken msg
+	reqDeviceName, err := GetContextItem(secTokenMsg, mdm.ReqSecTokenContextItemDeviceName)
+	if err != nil {
+		return fmt.Errorf("%s %v", error_tag, err)
+	}
+
+	// Getting the Enroll RequestVersion context information from the RequestSecurityToken msg
+	reqEnrollVersion, err := GetContextItem(secTokenMsg, mdm.ReqSecTokenContextItemRequestVersion)
+	if err != nil {
+		return fmt.Errorf("%s %v", error_tag, err)
+	}
+
+	// Getting the RequestVersion context information from the RequestSecurityToken msg
+	reqAppVersion, err := GetContextItem(secTokenMsg, mdm.ReqSecTokenContextItemApplicationVersion)
+	if err != nil {
+		return fmt.Errorf("%s %v", error_tag, err)
+	}
+
+	// Getting the EnrollmentType information from the RequestSecurityToken msg
+	reqEnrollType, err := GetContextItem(secTokenMsg, mdm.ReqSecTokenContextItemEnrollmentType)
+	if err != nil {
+		return fmt.Errorf("%s %v", error_tag, err)
+	}
+
+	// Getting the Windows Enrolled Device Information
+	enrolledDevice := &fleet.MDMWindowsEnrolledDevice{
+		MDMDeviceID:            reqDeviceID,
+		MDMHardwareID:          reqHWDevID,
+		MDMDeviceState:         mdm.MDMDeviceStateEnrolled,
+		MDMDeviceType:          reqDeviceType,
+		MDMDeviceName:          reqDeviceName,
+		MDMEnrollType:          reqEnrollType,
+		MDMEnrollUserID:        "", // No user information is available at this point
+		MDMEnrollProtoVersion:  reqEnrollVersion,
+		MDMEnrollClientVersion: reqAppVersion,
+		MDMNotInOOBE:           false,
+	}
+
+	if err := svc.ds.MDMWindowsInsertEnrolledDevice(ctx, enrolledDevice); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getWindowsDeviceID returns the Windows Device ID from the RequestSecurityToken message
+func GetContextItem(secTokenMsg *fleet.RequestSecurityToken, contextItem string) (string, error) {
+	reqHWDeviceID, err := secTokenMsg.GetContextItem(mdm.ReqSecTokenContextItemHWDevID)
```

Comment:
@marcosd4h just found out about this while implementing the activities 😅 

```suggestion
func GetContextItem(secTokenMsg *fleet.RequestSecurityToken, contextItem string) (string, error) {
	reqHWDeviceID, err := secTokenMsg.GetContextItem(contextItem)
```

I can also fix it in that branch if you prefer

Reply:
Great catch! I've just fixed this and improved the test to catch this scenario